src/credentials.rs

```diff
@@ -61,3 +62,21 @@ pub fn add_certificate(instance_name: &str, certificate: &str)
     task::block_on(write_credentials(&cred_path, &creds))?;
     Ok(())
 }
+
+pub async fn ask_update_credentials(
+    instance_name: &str, builder: &Builder
+) -> anyhow::Result<()> {
+    if builder.should_update_credential_file() {
+        let creds_path = path(instance_name)?;
+        if question::Confirm::new(format!(
+            "The format of the instance credential file at {} is outdated, \
+             update now?",
+            creds_path.display(),
+        )).ask()? {
```

Comment:
Using sync in async code, is always weird.

The problem here is that in `process.background_for` async block we handle signals differently. This is where async code signals are explicitly awaited and cancel other futures. So in this case `TERM` will not work during asking (can't cancel currently blocked future) and I'm not sure about `SIGINT` (it may fail input reader, but maybe not for all OSs).

And this is very bad since it can happen to wait for input in non-interactive environment, and then supervisor can't kill it normally by `TERM` and when it `KILL`s it, there is a hanging `process`.

Reply:
Yeah, I changed it to a blocking function.